src/dbnode/storage/shard.go

```diff
@@ -1867,15 +1867,21 @@ func (s *dbShard) Bootstrap(
 		info := result.Info
 		at := xtime.FromNanoseconds(info.BlockStart)
 		fs := s.FlushState(at)
-		if fs.WarmStatus != fileOpNotStarted {
-			continue // Already recorded progress
+
+		if fs.WarmStatus != fileOpSuccess {
+			s.markWarmFlushStateSuccess(at)
 		}
 
-		s.markWarmFlushStateSuccess(at)
-		// Cold version needs to get bootstrapped so that the 1:1 relationship between volume number
-		// and cold version is maintained and the volume numbers / flush versions remain monotonically
-		// increasing.
-		s.setFlushStateColdVersion(at, info.VolumeIndex)
+		// Cold version needs to get bootstrapped so that the 1:1 relationship
+		// between volume number and cold version is maintained and the volume
+		// numbers / flush versions remain monotonically increasing.
+		//
+		// Note that there can be multiple info files for the same block, for
```

Comment:
🤦‍♂ 

Sorry about that. Good catch...

Reply:
This merged separately in https://github.com/m3db/m3/pull/1819